### PR TITLE
feat: database schema overhaul, api rewrites and more

### PR DESCRIPTION
**Pull request by edustipend-root**
This PR does the following:

- Updates our data layer to use the newly provisioned MongoDB
- Rewrote API and service layers
- Proper logging around external calls
- Standardized API responses
- Updated session management logic to use JWTs instead of numeric digits

Not added:
- Updated tests

TODO:
- Clean up unused code and SQL-specific logic
- Need to update Swagger docs

### CODE REVIEW

File: constants/applicationStatus.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const ApplicationStatus = {
+  APPROVED: "Approved",
+  IN_REVIEW: "In Review",
+  RECEIVED: "Received",
+  UNAPPROVED: "Unapproved"
```
Comment (Chiemezuo) [inline constants/applicationStatus.js:5]: What about `DENIED: "Denied"`, so `Unapproved` doesn't get misinterpreted as `In Review` ?

File: utils/generateJwtToken.js [hunk 2]
```diff
@@ -1,23 +1,19 @@
 require("dotenv").config();
 const jwt = require("jsonwebtoken");
 
-const TOKEN_EXPIRATION_DEFAULT = "5d";
+const TOKEN_EXPIRATION_DEFAULT = "1h";
```
Comment (Chiemezuo) [inline utils/generateJwtToken.js:4]: Would it be better if we set this `TOKEN_EXPIRATION_DEFAULT` in the `.env` file, so we'd have a single goto place for all things related to token expirations?

File: utils/CSVFileParser.js
Comment (Chiemezuo) [inline utils/CSVFileParser.js:1]: I always wanted to ask what exactly this file does. 😅 
Maybe in our next meeting.

File: services/Authentication.js [hunk 3]
```diff
@@ -1,206 +1,83 @@
 const bcrypt = require("bcryptjs");
-const User = require("./User");
-const Token = require("./Token");
-const { randomSixDigits, validateEmail } = require("../utils/helper");
+const { findByUserName } = require("./User");
 const ErrorHandler = require("../utils/ErrorHandler");
 const models = require("../models");
+//TODO: Deprecate this
 const { generateJwtToken } = require("../utils/generateJwtToken");
```
Comment (Chiemezuo) [inline services/Authentication.js:6]: I suppose we can remove this...

**Review by Chiemezuo — COMMENTED** [review]
There was a lot to review. I hope I didn't miss much. I was more thorough with the modified files than the new ones. I'll still subsequently check them when doing proper testing.

File: utils/generateJwtToken.js (hunk 2 above)
Comment (edustipend-root) [reply utils/generateJwtToken.js:4]: I prefer to keep values that change based on environments in the `.env` file. This value won't change across environments so I think we should leave it in code.

I can move it the `constants.js` file in a subsequent PR.

File: utils/CSVFileParser.js
Comment (edustipend-root) [reply utils/CSVFileParser.js:1]: Haha, we used this when we had the issue of people not getting their email verified.

File: utils/CSVFileParser.js [hunk 4]
```diff
@@ -1,13 +1,22 @@
 const fs = require("fs");
 const { parse } = require("csv-parse");
+const { Mail, Token } = require("../services");
+const { randomSixDigits } = require("./helper");
```
[the remark this answers is not in the document]
Comment (edustipend-root) [reply utils/CSVFileParser.js:4]: Good catch, I'll delete this file.

File: scripts/parse_unverified_records.js
[the remark this answers is not in the document]
Comment (edustipend-root) [reply scripts/parse_unverified_records.js:1]: You don't need to worry about this, we are removing the file.

File: constants/applicationStatus.js (hunk 1 above)
Comment (edustipend-root) [reply constants/applicationStatus.js:5]: Let's address this in subsequent PRs.

Comment (Chiemezuo) [reply constants/applicationStatus.js:5]: Fair enough.

File: utils/generateJwtToken.js (hunk 2 above)
Comment (Chiemezuo) [reply utils/generateJwtToken.js:4]: I agree with this logic.

**Pull request merged**